providers/exoscale/exoscaleProvider.go

```diff
@@ -74,47 +96,69 @@ func (c *exoscaleProvider) GetZoneRecords(domain string) (models.Records, error)
 func (c *exoscaleProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
 	dc.Punycode()
 
+	domain, err := c.findDomainByName(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
 	ctx := context.Background()
-	records, err := c.client.GetRecords(ctx, dc.Name)
+	records, err := c.client.ListDNSDomainRecords(ctx, c.apiZone, *domain.ID)
 	if err != nil {
 		return nil, err
 	}
 
 	existingRecords := make([]*models.RecordConfig, 0, len(records))
 	for _, r := range records {
-		if r.RecordType == "SOA" || r.RecordType == "NS" {
+		record, err := c.client.GetDNSDomainRecord(ctx, c.apiZone, *domain.ID, *r.ID)
+		if err != nil {
+			return nil, err
+		}
+
+		if *record.Type == "SOA" || *record.Type == "NS" {
 			continue
 		}
-		if r.Name == "" {
-			r.Name = "@"
+		if *r.Name == "" {
+			t := "@"
+			record.Name = &t
 		}
-		if r.RecordType == "CNAME" || r.RecordType == "MX" || r.RecordType == "ALIAS" || r.RecordType == "SRV" {
-			r.Content += "."
+		if *record.Type == "CNAME" || *record.Type == "MX" || *record.Type == "ALIAS" || *record.Type == "SRV" {
+			t := *record.Content + "."
+			// for SRV records we need to aditionally prefix target with priority, which API handles as separate field.
+			if *record.Type == "SRV" && record.Priority != nil {
+				t = fmt.Sprintf("%s %s", strconv.FormatInt(*record.Priority, 10), t)
```

Comment:
Suggestion: Instead of `%s` and `strconv.FormatInt(*record.Priority, 10)` use `%d` and `*record.Priority` ?